### DISCUSSION TITLE
Get error when I try to delete element of array

### DISCUSSION THREAD

**Discussion by shakhban777** [mobxjs/mobx-state-tree#1790] [Q&A]
Hello everyone, I'm new in MST and I got issue with this amazing state manager.
I create Data Model
![image](https://user-images.githubusercontent.com/64259612/133939681-acd45b22-afbf-4b32-acaf-26be9cab4999.png)

And use this model in another one
![image](https://user-images.githubusercontent.com/64259612/133939770-38ce1f1d-8216-4bf6-a1b5-7dee945576ad.png)

And when I'm trying to delete any items of OperationsStore.data elements I'm getting error
  
![image](https://user-images.githubusercontent.com/64259612/133939792-552b7a98-956d-4df2-be1e-c0b8cda79880.png)

Could you say me please, how can fix this error?  Thanks.

**Comment by EmilTholin** [comment]
Hi @shakhban777!

It's hard to say from the code in your question what might be going wrong. Do you think you could create a reproducible example in e.g. [CodeSandbox](https://codesandbox.io/s/) that gives rise to your error, and possibly include your code as text instead of as images?

**Reply by shakhban777** [reply]
https://codesandbox.io/s/elegant-flower-vv8ld?file=/src/stores/OperationsStore.ts
Test username - Rinaz and password - 123456.
MST is in the file OperationsStore.ts

Delete container - https://codesandbox.io/s/elegant-flower-vv8ld?file=/src/pages/Home/containers/DeleteTransaction/index.tsx

Api - https://codesandbox.io/s/elegant-flower-vv8ld?file=/src/services/ApiService/index.ts

Where I use Delete container:

- https://codesandbox.io/s/elegant-flower-vv8ld?file=/src/pages/Home/containers/TableContainer/TableButtons/index.tsx
- https://codesandbox.io/s/elegant-flower-vv8ld?file=/src/pages/Home/containers/SidebarWithCheckedTransactions/index.tsx

In codesandbox I got some warnings, don't worry about it. I want to fix the error when I'm trying to delete any item (or group of items via checkboxes)
